server/src/main/java/org/elasticsearch/index/mapper/PassThroughObjectMapper.java

```diff
@@ -61,42 +81,66 @@ public PassThroughObjectMapper build(MapperBuilderContext context) {
                 enabled,
                 dynamic,
                 buildMappers(context.createChildContext(name(), timeSeriesDimensionSubFields.value(), dynamic)),
-                timeSeriesDimensionSubFields
+                timeSeriesDimensionSubFields,
+                supersededBy
             );
         }
     }
 
     // If set, its subfields are marked as time-series dimensions (for the types supporting this).
     private final Explicit<Boolean> timeSeriesDimensionSubFields;
 
+    private final Set<String> supersededBy;
+
     PassThroughObjectMapper(
         String name,
         String fullPath,
         Explicit<Boolean> enabled,
         Dynamic dynamic,
         Map<String, Mapper> mappers,
-        Explicit<Boolean> timeSeriesDimensionSubFields
+        Explicit<Boolean> timeSeriesDimensionSubFields,
+        Set<String> supersededBy
```

Comment:
Maybe an alternative way to deal with conflicts could be having a `priority` mapping attribute instead? (range from: 0 to max integer)

If there are multiple passthrough fields then a priority attribute is required on all passthrough fields. If there're passthough fields with the same priority we fail with a validation error. If multiple passthrough fields match we pick the passthough field with the highest priority.


I think this could simplify to validation logic / logic in `FieldTypeLookup`?
Note that index templates have a similar notion of priority.

Reply:
The problem with this is that a user may start with one passthrough field, then add another one, then notice that there are conflicts. I don't think we should require users to have a finalized list of pass-through objects from the get go, @felixbarny too.

Reply:
Without having thought very deeply about this, I was also thinking of a numeric `priority` mapping parameter. But need to read more to get a better understanding of what `supersededBy` is doing.

Reply:
For OTel mappings, the precedence (from highest to lowest) would look something like this:

* Top-level/non passthrough fields
* `resource.attributes.*`
* `scope.attributes.*`
* `attributes.*`

To me, it seems that could be achieved with a simple numeric `priority` mapping parameter, or, at least with a single value for `supersededBy`.

Reply:
Switched to using priorities. An issue that I see is what happens with existing pass-through objects missing priorities.. I can check for the existence of aliases and bypass the check for priorities, but that's not great either.. Thoughts?

Reply:
I think we haven't really used the passthrough field type outside of testing, so this breaking change is probably fine. Alternatively, we could not require passthrough fields to have unique priorities and fall back to lexicographical ordering if the priority is the same.

Reply:
Let's make priority required. This field type is undocumented and not ga-ed. This field type is only used for cross team testing at the moment.